optimum/exporters/neuron/__main__.py

```diff
@@ -14,141 +14,230 @@
+
 
 if is_neuronx_available():
     from ...commands.export.neuronx import parse_args_neuronx as parse_args_neuron  # noqa: F811
 
     NEURON_COMPILER = "Neuronx"
-    NEURON_COMPILER_TYPE = "neuronx-cc"
-    NEURON_COMPILER_VERSION = get_neuronxcc_version()
 
 
 logger = logging.get_logger()
 logger.setLevel(logging.INFO)
 
 
-def main():
-    parser = ArgumentParser(f"Hugging Face Optimum {NEURON_COMPILER} exporter")
-
-    parse_args_neuron(parser)
+def infer_compiler_kwargs(args: argparse.Namespace):
+    # infer compiler kwargs
+    auto_cast = None if args.auto_cast == "none" else args.auto_cast
+    auto_cast_type = None if auto_cast is None else args.auto_cast_type
+    compiler_kwargs = {"auto_cast": auto_cast, "auto_cast_type": auto_cast_type}
+    if hasattr(args, "disable_fast_relayout"):
+        compiler_kwargs["disable_fast_relayout"] = getattr(args, "disable_fast_relayout")
+    if hasattr(args, "disable_fallback"):
+        compiler_kwargs["disable_fallback"] = getattr(args, "disable_fallback")
 
-    # Retrieve CLI arguments
-    args = parser.parse_args()
-    args.output = args.output.joinpath("model.neuron")
+    return compiler_kwargs
 
-    if not args.output.parent.exists():
-        args.output.parent.mkdir(parents=True)
 
-    # Infer the task
-    task = args.task
+def infer_task(task: str, model_name_or_path: str):
     if task == "auto":
         try:
-            task = TasksManager.infer_task_from_model(args.model)
+            task = TasksManager.infer_task_from_model(model_name_or_path)
         except KeyError as e:
             raise KeyError(
                 "The task could not be automatically inferred. Please provide the argument --task with the task "
                 f"from {', '.join(TasksManager.get_all_tasks())}. Detailed error: {e}"
             )
+        except RequestsConnectionError as e:
+            raise RequestsConnectionError(
+                f"The task could not be automatically inferred as this is available only for models hosted on the Hugging Face Hub. Please provide the argument --task with the relevant task from {', '.join(TasksManager.get_all_tasks())}. Detailed error: {e}"
+            )
+    return task
+
+
+def normalize_input_shapes(task: str, args: argparse.Namespace):
+    # get input shapes
+    if task == "stable-diffusion":
+        mandatory_shapes = {
+            name: getattr(args, name, None)
+            for name in getattr(inspect.getfullargspec(build_stable_diffusion_components_mandatory_shapes), "args")
+        }
+        input_shapes = build_stable_diffusion_components_mandatory_shapes(**mandatory_shapes)
+    else:
+        model = TasksManager.get_model_from_task(
+            task, args.model, framework="pt", cache_dir=args.cache_dir, trust_remote_code=args.trust_remote_code
+        )
+        neuron_config_constructor = TasksManager.get_exporter_config_constructor(
+            model=model, exporter="neuron", task=task
+        )
+        del model
```

Comment:
I can be done without loading the model no?

Reply:
Yeah I can infer that through `model_type` of the model config as well. It seems better than loading the model, or do you have any other ideas?

Reply:
I think it's much better yes!